webkit2/webkit2.uri-scheme-request.lisp

```diff
@@ -0,0 +1,105 @@
+;;; webkit2.uri-scheme-request.lisp --- bindings for WebKitURISchemeRequest
+
+;; This file is part of cl-webkit.
+;;
+;; cl-webkit is free software; you can redistribute it and/or modify
+;; it under the terms of the MIT license.
+;; See `COPYING' in the source distribution for details.
+
+;;; Code:
+
+(in-package #:webkit2)
+
+(define-webkit-class "WebKitURISchemeRequest" () ())
+
+(define-g-object-class "GInputStream" g-input-stream
+    (:export nil
+     :type-initializer "g_input_stream_get_type")
+    ())
+(define-g-object-class "GMemoryInputStream" g-memory-input-stream
+    (:superclass g-input-stream
+     :export nil
+     :type-initializer "g_memory_input_stream_get_type")
+    ())
+
+(defcfun "webkit_uri_scheme_request_get_scheme" :string
+  (request (g-object webkit-uri-scheme-request)))
+(export 'webkit-uri-scheme-request-get-scheme)
+
+(defcfun "webkit_uri_scheme_request_get_uri" :string
+  (request (g-object webkit-uri-scheme-request)))
+(export 'webkit-uri-scheme-request-get-uri)
+
+(defcfun "webkit_uri_scheme_request_get_path" :string
+  (request (g-object webkit-uri-scheme-request)))
+(export 'webkit-uri-scheme-request-get-path)
+
+(defcfun "webkit_uri_scheme_request_get_web-view" (g-object webkit-web-view)
+  (request (g-object webkit-uri-scheme-request)))
+(export 'webkit-uri-scheme-request-get-web-view)
+
+(defcfun "webkit_uri_scheme_request_finish" :void
+  (request (g-object webkit-uri-scheme-request))
+  (stream (g-object g-memory-input-stream))
+  (stream-length :long)
+  (content-type :string))
+(export 'webkit-uri-scheme-request-finish)
+
+(defcfun ("webkit_uri_scheme_request_finish_error" %webkit-uri-scheme-request-finish-error) :void
+  (request (g-object webkit-uri-scheme-request))
+  (g-error :pointer))
+
+(defun webkit-uri-scheme-request-finish-error (request error-string)
+  (%webkit-uri-scheme-request-finish-error
+   request (glib::%g-error-new-literal
+            +webkit-plugin-error+
+            4
```

Comment:
What's this magic number?

Reply:
This is the `:webkit-plugin-error-connection-cancelled`. It's used this way because `glib::%g-error-new-literal` does not accept keywords as enum elements, so we need to use integers.

I'll add the comment explaining this.

Reply:
Better, define a CL constant, e.g. `+webkit-plugin-error-connection-cancelled+`.

Reply:
Done, see 2fa25f6 :)